web3/middleware/stalecheck.py

```diff
@@ -0,0 +1,55 @@
+import datetime
+import time
+
+
+SKIP_STALECHECK_FOR_METHODS = set([
+    'eth_getBlockByHash',
+    'eth_getBlockByNumber',
+    'eth_getBlockTransactionCountByHash',
+    'eth_getBlockTransactionCountByNumber',
+    'eth_getTransactionByBlockHashAndIndex',
```

Comment:
Why are the `getTransactionByBockXXXXAndIndex` methods included?

Reply:
I waffled a bit on this locally, so am very flexible.

I figured any call that specifically references the block number/hash seemed safe, since it will fail anyway if the block isn't found. This seemed like the cleanest way to define the methods to blacklist. The next-best alternative seemed to be just the "getBlock*" methods, but that seemed more like patching my bug than logically sound.

Reply:
So here's how I'm thinking about this.

The job of this middleware is explicitly cause failures when the chain goes out of sync.  In my little web3 world this is intended to allow me to differentiate between failure caused by a non-existent block hash and failure caused by being out of sync. 

But that raises the question of whether the stale checking middleware should only throw an error *if* the result is `None`...  Honestly not sure.  Your call.

Reply:
Ah, interesting.

For ENS, at least, it's bad to get valid but old data (you might be resolving to an old address, for example). So it's best to raise an exception no matter what on old data.

It seems like the least contentious option then is to switch to only "getBlockBy*", and arguably, only if the parameter is `'latest'`.